### PR TITLE
build: Koa support Node.js@10+

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
Current Node.js engines inside the pkg.js are in EOF.

| Node.js | End Of Life |
| ---------| -------------|
| 4.x.x      | 2018-04-30 |
| 6.x.x      | 2019-04-30 |
| 7.x.x      | 2017-06-30 |
| 8.x.x      | 2019-12-31 |
| 8.x.x      | 2019-12-31 |

> refs: https://github.com/nodejs/Release

When these PR are accepted. I will open a new one to update docs.

fix: #1051

### CODE REVIEW

**Comment by 3imed-jaberi** [comment]
@dead-horse @jonathanong,
Please, can you check this !! it's related to #1051.

**Comment by miwnwski** [comment]
I'm ok with this as long as we do this as long as we bump major.

File: package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "supertest": "^3.1.0"
   },
   "engines": {
-    "node": "^4.8.4 || ^6.10.1 || ^7.10.1 || >= 8.1.4"
```
Comment (dead-horse) [inline package.json:72]: We should break the promise that we support these node versions in current version even they were EOF. we can change this in the next major version but not now.

Comment (anlexN) [reply package.json:72]: @dead-horse you can not do anything. you don't know when to release V3 too.
@3imed-jaberi can we together rewrite koa with LTS and current node? 😄

Comment (miwnwski) [reply package.json:72]: @anlexN Koa is MIT licensed so you are free to fork and do what ever you want with it. However, please be respectful to all contributors. We (Koa maintainers of different ranks) are in agreement that a new version of Koa isn't warranted unless there's significant changes/improvements to the library. You may disagree with the choices made - but that's no reason to be disrespectful. You started writing tickets a few days ago, we've been maintaining Koa for years.

Most of the things you've raised are breaking changes. We take semver very seriously here and have a rather large ecosystem to consider before introducing breakage. There will always be things to improve, we are always open to debate changes, PRs are always welcome. 

An underlined image with a passive aggressive comment isn't fun for us. Keep it respectful and joyful for all parties involved. A happy ticket, and preferably PR, is more likely to be taken seriously.

Comment (anlexN) [reply package.json:72]: can we make a new branch focused on new idea? i need this branch very much, i hope many people like it too, and so many pull request sound very wonderful but you don't like merge.

Comment (miwnwski) [reply package.json:72]: You can fork the repo and create as many branches as you need.

Comment (anlexN) [reply package.json:72]: i hope you make, you are official, please?

Comment (anlexN) [reply package.json:72]: @3imed-jaberi please review [anlexN/koa](https://github.com/anlexN/koa), we can together do it.

Comment (3imed-jaberi) [reply package.json:72]: Hi @dead-horse, @miwnwski, @anlexN 👋 .. 
Sorry to be absent for a month due to my graduation ... I believe that working on the next release (koa@3.x.x) and planning for it with somethings like improve perf. is more important than rewrite with typescript. !!

**Comment by faddat** [comment]
ah-- so Koa does not currently work with, for example, node 14?

Suppose I have an app...... how does one go about replacing Koa?   am I doomed?

**Comment by 3imed-jaberi** [comment]
@faddat, don't worry, you can use `Koa` with any Node.js version.

This PR it's about CI strategy and the way to write the code, I propose to drop support for old versions natively and support only the life versions.

**Comment by anlexN** [comment]
expect koa 3 very much

**Comment by faddat** [comment]
@3imed-jaberi Thanks very much for explaining 🙏 .

Guess I will need to sort through my other dependencies, something is not liking new nodejs....

I am streamlining and modernizing a rather aged Koa app.

**Pull request closed without merging**